lapce-data/src/palette.rs

```diff
@@ -345,6 +345,7 @@ pub struct PaletteData {
     pub input: String,
     pub cursor: usize,
     pub index: usize,
+    pub has_nonzero_default_index: bool,
```

Comment:
Can we turn `index` into an `Option` instead?

Reply:
Index is used regardless of whether there's a default selection or not, so making it an Option would (imo) make everything a little harder to read and less clear.
I mean, even if there's no previous value, the item at index 0 defaults to being selected, right?

Reply:
Thanks :)